tests/e2e/framework/kubernetes.go

```diff
@@ -0,0 +1,230 @@
+	mixerHubDefault    string
+	mixerTagDefault    string
+	managerHubDefault  string
+	managerTagDefault  string
+)
+
+type KubeInfo struct {
+	Namespace        string
+	NamespaceCreated bool
+	Hub              string
+	Tag              string
+	MixerImage       string
+	ManagerImage     string
+	CaImage          string
+	ProxyHub         string
+	ProxyTag         string
+	IstioCtlUrl      string
+	IstioCtl         string
+	Verbosity        int
+
+	TmpDir  string
+	YamlDir string
+}
+
+func NewKubeInfo(tmpDir, runId string) *KubeInfo {
+	parseVersion()
+	hubDefault := "gcr.io/istio-testing"
+	tagDefault := "b121a1e169365865e01a9e6eea066a34a29d9fd1"
+
+	hub := flag.String("hub", hubDefault, "Docker hub")
+	tag := flag.String("tag", tagDefault, "Docker tag")
+	namespace := flag.String("n", runId, "Namespace to use for testing (empty to create/delete temporary one)")
+	mixerImage := flag.String("mixer", mixerHubDefault+"/mixer:"+mixerTagDefault, "Mixer image")
+	managerImage := flag.String("manager", managerHubDefault+"/manager:"+managerTagDefault, "Manager image")
+	caImage := flag.String("ca", "", "Ca image")
+	proxyHub := flag.String("proxy_hub", managerHubDefault, "proxy hub")
+	proxyTag := flag.String("proxy_tag", managerTagDefault, "proxy tag")
+	istioCtlUrl := flag.String("istioctl_url", istioCtlUrlDefault, "URL to download istioctl")
+	verbose := flag.Bool("verbose", false, "Debug level noise from proxies")
+
+	var verbosity int
+	if *verbose {
+		verbosity = 3
+	} else {
+		verbosity = 2
+	}
+
+	return &KubeInfo{
+		Hub:              *hub,
+		Tag:              *tag,
+		Namespace:        *namespace,
+		MixerImage:       *mixerImage,
+		ManagerImage:     *managerImage,
+		CaImage:          *caImage,
+		ProxyHub:         *proxyHub,
+		ProxyTag:         *proxyTag,
+		IstioCtlUrl:      *istioCtlUrl,
+		Verbosity:        verbosity,
+		NamespaceCreated: false,
+		TmpDir:           tmpDir,
+		YamlDir:          tmpDir + "/yaml",
+	}
+}
+
+func (k *KubeInfo) Setup() error {
+	if err := util.CreateNamespace(k.Namespace); err != nil {
+		glog.Error("Failed to create namespace.")
+		return err
+	}
+	k.NamespaceCreated = true
+	if err := k.deployIstio(); err != nil {
+		glog.Error("Failed to deployIstio.")
+		return err
+	}
+
+	glog.Info("Kubernetes setup finished.")
+	return nil
+}
+
+func (k *KubeInfo) Teardown() error {
+	if k.NamespaceCreated {
+		if err := util.DeleteNamespace(k.Namespace); err != nil {
+			return err
+		}
+		k.NamespaceCreated = false
+		glog.Infof("Namespace %s deleted", k.Namespace)
+	}
+	glog.Info("Uploading log remotely")
+	glog.Flush()
+	return nil
+}
+
+// Deploy istio modules
+func (k *KubeInfo) deployIstio() error {
+	if err := os.Mkdir(k.YamlDir, os.ModeDir|os.ModePerm); err != nil {
+		return err
+	}
+	usr, eUser := user.Current()
+	if eUser != nil {
+		return eUser
+	}
+	homeDir := usr.HomeDir
+
+	// Download Istioctl binary
+	util.WebDownload(k.TmpDir+"/istioctl", k.IstioCtlUrl+"/istioctl-linux")
```

Comment:
Move this to another library. It is hard to understand why you are doing this here.